### PR TITLE
RDM-3142 CaseRoles StartEvent story

### PR DESCRIPTION
**Pull request by tzarsmango**
https://tools.hmcts.net/jira/browse/RDM-3142

This PR is about applying the CaseRoles that are granted via the `case_users` table.  The start event and create event requests
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/AuthorisedCreateEventOperation.java [hunk 1]
```diff
@@ -113,6 +135,20 @@ private CaseDetails verifyReadAccess(CaseType caseType, Set<String> userRoles, C
         return caseDetails;
     }
 
+    private Set<String> getCaseRoles(String caseId) {
+        return caseUserRepository
+            .findCaseRoles(Long.valueOf(caseId), userRepository.getUserId())
+            .stream()
+            .collect(Collectors.toSet());
```
Comment (vlaurin) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/AuthorisedCreateEventOperation.java]: This logic seems to be duplicated 3 times in this PR, might be good to extract it into the case role service.

Comment (tzarsmango) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/AuthorisedCreateEventOperation.java]: 👍

File: src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CaseRoleService.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+
+@Service
+public class CaseRoleService {
```
Comment (vlaurin) [inline src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CaseRoleService.java]: Couldn't this be merged with the already existing `CaseAccessService`? (https://github.com/hmcts/ccd-data-store-api/blob/master/src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessService.java)

It's got the same dependencies, and is responsible for case access, which is a superset of case roles.

Comment (tzarsmango) [reply src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CaseRoleService.java]: done

File: src/main/java/uk/gov/hmcts/ccd/data/caseaccess/GlobalCaseRole.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package uk.gov.hmcts.ccd.data.caseaccess;
+ package uk.gov.hmcts.ccd.data.caseaccess;
```
Comment (vlaurin) [inline src/main/java/uk/gov/hmcts/ccd/data/caseaccess/GlobalCaseRole.java]: Que? :)

Comment (tzarsmango) [reply src/main/java/uk/gov/hmcts/ccd/data/caseaccess/GlobalCaseRole.java]: who knows why? :)

**Comment by tzarsmango** [comment]
After the refactoring to keep Sonar happy, created a tech debt task for completion after the other task in this epic https://tools.hmcts.net/jira/browse/RDM-3614

**Pull request merged**